### PR TITLE
update to jar-25 and incorporated text about require_signed_request_object

### PR DESCRIPTION
**Pull request by tlodderstedt**
please give it a read

### CODE REVIEW

File: main.md [hunk 1]
```diff
@@ -518,10 +519,11 @@ Specification Document(s):
 
    * Update Resource Indicators reference to the somewhat recently published RFC 8707
    * Added metadata in support of pushed authorization requests only feature
-   * update to comply with draft-ietf-oauth-jwsreq-21, which requires `client_id` in the authorization request in addition to the `request_uri`
+   * Update to comply with draft-ietf-oauth-jwsreq-21, which requires `client_id` in the authorization request in addition to the `request_uri`
    * Clarified timing of request validation
    * Add some guidance/options on the request URI structure
    * Add the key used in the request object example so that a reader could validate or recreate the request object signature
+   * Update to draft-ietf-oauth-jwsreq-25 and added not regarding `require_signed_request_object`
```
Comment (b---c) [inline main.md:526]:
```suggestion
   * Update to draft-ietf-oauth-jwsreq-25 and added note regarding `require_signed_request_object`
```

File: main.md [hunk 2]
```diff
@@ -235,6 +235,8 @@ For an error the authorization server sets an appropriate HTTP status code and M
 
 If the authorization server sets an error code, it SHOULD be one of the defined codes for the token endpoint in Section 5.2 or for the authorization endpoint in Sections 4.1.2.1 and 4.2.2.1 of [@!RFC6749], or by an OAuth extension if one is involved in the initial processing of authorization request that was pushed. Since initial processing of the pushed authorisation request doesn't involve resource owner interaction, error codes related to user interaction, such as `consent_required` defined by [@!OIDC], are not returned.
 
+If the client is required to use signed request objects, either by authorization server or client policy (see [@!I-D.ietf-oauth-jwsreq], sections 9.2 and 9.3), the authorization server MUST only accept requests complying with the definition given in (#request_parameter) and MUST refuse any other request with HTTP status code 400 and error code `invalid_request`.
```
Comment (b---c) [inline main.md:238]:
```suggestion
If the client is required to use signed request objects, either by authorization server or client policy (see [@!I-D.ietf-oauth-jwsreq], section 10.5), the authorization server MUST only accept requests complying with the definition given in (#request_parameter) and MUST refuse any other request with HTTP status code 400 and error code `invalid_request`. 
```

File: main.md (hunk 1 above)
Comment (b---c) [reply main.md:526]: just a typo: not -> note

File: main.md (hunk 2 above)
Comment (b---c) [reply main.md:238]: JAR §10.5 seems more appropriate than the IANA section to reference.

**Review by davidgtonge — APPROVED** [review]

**Pull request merged**